s3-backed-core/src/main/java/com/bakdata/kafka/AbstractS3BackedConfig.java

```diff
@@ -156,11 +171,28 @@ private boolean enablePathStyleAccess() {
     private Optional<AWSCredentialsProvider> getCredentialsProvider() {
         final String accessKey = this.getPassword(S3_ACCESS_KEY_CONFIG).value();
         final String secretKey = this.getPassword(S3_SECRET_KEY_CONFIG).value();
-        if (isEmpty(accessKey) || isEmpty(secretKey)) {
-            return Optional.empty();
-        } else {
+        
+        if (!isEmpty(accessKey) || !isEmpty(secretKey)) {
             final AWSCredentials credentials = new BasicAWSCredentials(accessKey, secretKey);
             return Optional.of(new AWSStaticCredentialsProvider(credentials));
         }
+
+        final String roleExternalId = this.getString(S3_ROLE_EXTERNAL_ID_CONFIG);
```

Comment:
Are these three values anything secret or is string fine?

Reply:
No, they aren't secret and Strings should be fine.